index.js

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env node
+require("dotenv").config();
+const axios = require("axios");
+
+// Usage from the command line:
+// npm install
+// node index sourceChannel targetChannel
+// e.g. node index general dev-chatter
+
+// Note: Do not include the hashtag in the channel name. Though I did account for stripping the hashtag away, process.argv returns undefined if the hashtag is present and I'm unsure of why.
+
+// Store Slack access token in a .env file in the root folder
+// You can ping a JDSB admin on Slack for the OAuth token, or request to become a collaborator on the app.
+// The token for an app can be found by selecting the app from api.slack.com/apps and navigating to "OAuth & Permissions" page after it has been installed to a workspace.
+const accessToken = process.env.SLACK_ACCESS_TOKEN;
+
+async function getConversationId(conversation) {
+
+  // Trim the pound sign if it was provided in the conversation argument
+  if (conversation.startsWith('#')) {
+    conversation = conversation.slice(1)
+  }
+
+  // Destructuring to pull out the channels result from the response
+  const { data: { channels } } = await axios.get(`https://slack.com/api/conversations.list?token=${accessToken}&pretty=1`)
+
+  // Find the channel with matching conversation name and pull out its ID
+  const conversationId = channels.find(chan => chan.name === conversation).id
+
+  return conversationId
+}
+
+async function getConversationMembers(conversationId) {
+
+  async function queryForMembers(membersCollection = [], nextCursor = '') {
+    let allConversationMembers = [];
+
+    // Base case: members exist, no next cursor, we've retrieved all the members
+    if (membersCollection.length > 0 && !nextCursor) {
```

Comment:
membersCollection.length > 0 is extraneous. You should just return whatever list you have. The n-1th recursion should be able to handle an empty list.

Also, your base case will never trigger, if the totalNumberOfMembers % pageSize === 0. This would be a bug.

Reply:
Also, just fyi, just because you CAN do recursion, doesn't mean you should. You're filling up your callstack unnecessarily here with every function call. There's a way to do this in a do-while loop.

Reply:
Not sure I follow here. 

The check is `If the array isn't empty` (which indicates we've gathered some members -- it's not enough if its just the original empty array existing), and `There is no next cursor provided` (which indicates there are no more pages to check), then we've paged through all the results and gotten all the members, so stop recursion and return them.

Also not sure why base case wouldn't trigger under that circumstance -- your the math guy, but isn't that the same as saying totalNumberOfMembers === pageSize? 200 members, default page size is 200, remainder 0. In that case, membersCollection has length, and no nextCursor is provided -- why wouldn't it trigger?

Reply:
@bhurstGH for this question, I'm going to suggest writing test cases for this to help you answer your questions. Basically, nextCursor being empty is sufficient to determine whether we should stop processing.

Reply:
re. member list size...and so is 0 mod pageSize === 0, if Slack ever sends that back. They probably don't, but why risk it when the other check is extraneous?

Reply:
Talked with @bhurstGH offline. We agreed this could use a refactor, but we'll save it for when this thing moves into a Slack App. It won't help anything to refactor and productionalize this right now without a good test method.